src/mmrelay/db_runtime.py

```
@@ -411,22 +423,74 @@ def executor_func() -> Any:
+        self, worker_future: Future[Any]
+    ) -> None:
+        """
+        Surface write worker failures that complete after caller cancellation.
+        """
+
+        def _log_future_error(resolved_future: Future[Any]) -> None:
+            try:
+                resolved_future.result()
+            except (asyncio.CancelledError, ConcurrentCancelledError):
+                pass
+            except BaseException:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `BaseException`

(BLE001)